common/app/uiComponent/core/JavascriptEngine.scala

```diff
@@ -0,0 +1,25 @@
+package uiComponent.core
+
+import java.io.Reader
+import javax.script._
+
+import jdk.nashorn.api.scripting.JSObject
+
+import scala.util.Try
+
+class JavascriptEngine() {
+  import JavascriptEngine._
+
+  private lazy val bindings = shared.createBindings()
```

Comment:
`JavascriptEngine` was already already a `class` before, to make sure bindings were not shared. 
Not sharing the context/bindings seems to be the only tricky bit, thanks for making sure I am doing the right things